### ISSUE TITLE
Question Snapshots

### ISSUE THREAD

**Issue by starflighter** [sdumetz/aptly-web-ui#5]
Hi,
is it possible to take and edit aptly-snapshots in this web-ui?
That would be great :-)

Thanks
Yvonne

**Comment by sdumetz** [comment]
Nope. This UI is currently not snapshot-aware. It's definitely something I'd like to add though.

**Comment by starflighter** [comment]
That would be awsome!
If I can help, just say it. But I can't write Java Script, only HTML and Shell.

**Comment by sdumetz** [comment]
Right now, I'm at the "define how it would work and how it would look" step, so you can definitely help with no JS knowledge.

What would be for you the minimal function set for this feature to be usable and to know how it maps in aptly API calls?

**Comment by starflighter** [comment]
I haven't worked long with aptly yet. I need some time to look at the api and some use-cases for the snapshots. So I hope I can tell you the next 2 weeks more....

**Comment by starflighter** [comment]
Sorry for the delay.
I need more time, a lot of other stuff to do at work :-)...
The minimal set is to get, add and delete the snapshots. 
Is it possible to get a list of all the packages in a repo and you can choose which you like in the new snapshot? That would be a lot of packages... the performance would be probably horrible.

**Comment by sdumetz** [comment]
I implemented getting the snapshots in the latest commits. Deleting them should also be straightforward.

However I'm not yet sure about how snapshot creation should be handled? Maybe a simple button `Create snapshot from "reponame"` in `/ui/snapshots` would do for most use cases?

**Comment by starflighter** [comment]
That would be enough I think.

**Comment by starflighter** [comment]
I tried the new version of the web-ui. It looks good.
![grafik](https://user-images.githubusercontent.com/23389327/30205306-b11c488e-9488-11e7-8c88-3a9eef5b7a2a.png)

But can you change the date of the snapshot to format yyyy-mm-dd-hh-mm-ss? That is more readable.

**Comment by sdumetz** [comment]
Will do. Thanks for testing.

**Comment by sdumetz** [comment]
I've added a (crude) snapshot creation tool in v0.2.2 that should work.

It's useless as long as there is no way to restore/delete snapshots but hey, it's a start ;-)

**Comment by starflighter** [comment]
Hi,
I tried the version 02.2 today - it looks good...
![grafik](https://user-images.githubusercontent.com/23389327/30743128-a7c7cd02-9f9c-11e7-8a6c-5fc91293102d.png)

Just for your information I set it up on a Ubuntu 16.04.3 LTS with nginx/1.10.3 (Ubuntu). No problems here :-)
Next step is the action "delete snapshot" ?

**Comment by sdumetz** [comment]
I've an important rework on the publish API that's needed to support this. 
As soon as it's done (next minor version), I'll be able to handle snapshot removal and publication easily.
